src/hotspot/share/runtime/frame.cpp

```diff
@@ -1142,6 +1142,7 @@ void frame::interpreter_frame_verify_monitor(BasicObjectLock* value) const {
 
 #ifndef PRODUCT
 void frame::describe(FrameValues& values, int frame_no) {
+  ResourceMark rm;
```

Comment:
Not sure this works. RA allocated memory escapes the function in FrameValues::describe() (via FrameValue::description):

```
void FrameValues::describe(int owner, intptr_t* location, const char* description, int priority) {
  FrameValue fv;
  fv.location = location;
  fv.owner = owner;
  fv.priority = priority;
  fv.description = NEW_RESOURCE_ARRAY(char, strlen(description) + 1);
  strcpy(fv.description, description);
  _values.append(fv);
}
```

Reply:
Okay thanks , we have to take escaping memory indeed into consideration .
 I  think I should omit the frame_zero.cpp  change.
In frame.cpp I could use a local buffer (any idea  how large?)  and the name_and_sig_as_C_string-version taking a pre-allocated buffer :   name_and_sig_as_C_string(buffer, buf_len)   .
Then there is not need to add the  ResourceMark , correct ?